src/include/smash/outputinterface.h

```diff
@@ -21,6 +21,7 @@
 #include "macros.h"
 
 namespace smash {
+static constexpr int LOutput = LogArea::Output::id;
```

Comment:
After these two fixes, the compilation is successful if Rivet is not included.
When Rivet is included, it fails at the end with this message:
(...)
[100%] Building CXX object src/CMakeFiles/objlib.dir/hepmcinterface.cc.o
[100%] Building CXX object src/CMakeFiles/objlib.dir/rivetoutput.cc.o
/home/g/SMASH/smash-cholmcc/src/rivetoutput.cc: In member function ‘virtual void smash::RivetOutput::at_eventend(const smash::Particles&, int32_t, const smash::EventInfo&)’:
/home/g/SMASH/smash-cholmcc/src/rivetoutput.cc:172:19: error: cannot convert ‘HepMC3::GenEvent’ to ‘const HepMC::GenEvent&’
  172 |     proxy()->init(event_);
      |                   ^~~~~~
      |                   |
      |                   HepMC3::GenEvent
In file included from /home/g/SMASH/smash-cholmcc/src/include/smash/rivetoutput.h:12,
                 from /home/g/SMASH/smash-cholmcc/src/rivetoutput.cc:10:
/home/g/rivet_analysis/rivet_build/include/Rivet/AnalysisHandler.hh:224:31: note:   initializing argument 1 of ‘void Rivet::AnalysisHandler::init(const HepMC::GenEvent&)’
  224 |     void init(const GenEvent& event);
      |               ~~~~~~~~~~~~~~~~^~~~~
/home/g/SMASH/smash-cholmcc/src/rivetoutput.cc:177:26: error: no matching function for call to ‘Rivet::AnalysisHandler::analyze(HepMC3::GenEvent&)’
  177 |   proxy()->analyze(event_);
      |                          ^
In file included from /home/g/SMASH/smash-cholmcc/src/include/smash/rivetoutput.h:12,
                 from /home/g/SMASH/smash-cholmcc/src/rivetoutput.cc:10:
/home/g/rivet_analysis/rivet_build/include/Rivet/AnalysisHandler.hh:230:10: note: candidate: ‘void Rivet::AnalysisHandler::analyze(const HepMC::GenEvent&)’
  230 |     void analyze(const GenEvent& event);
      |          ^~~~~~~
/home/g/rivet_analysis/rivet_build/include/Rivet/AnalysisHandler.hh:230:34: note:   no known conversion for argument 1 from ‘HepMC3::GenEvent’ to ‘const HepMC::GenEvent&’
  230 |     void analyze(const GenEvent& event);
      |                  ~~~~~~~~~~~~~~~~^~~~~
/home/g/rivet_analysis/rivet_build/include/Rivet/AnalysisHandler.hh:236:10: note: candidate: ‘void Rivet::AnalysisHandler::analyze(const HepMC::GenEvent*)’
  236 |     void analyze(const GenEvent* event);
      |          ^~~~~~~
/home/g/rivet_analysis/rivet_build/include/Rivet/AnalysisHandler.hh:236:34: note:   no known conversion for argument 1 from ‘HepMC3::GenEvent’ to ‘const HepMC::GenEvent*’
  236 |     void analyze(const GenEvent* event);
      |                  ~~~~~~~~~~~~~~~~^~~~~

It's clear what is (most likely) the problem, but right now I don't have the time to look more into detail and fix it, I will come back to it shortly, but maybe meanwhile you can already adjust the issue.

Reply:
I think that you are probably missing some commits wrt log I'd and source location. The third problem is because you have built Rivet against HepMC and not HepMC3. 

Note, I am not part of the SMASH team and I do not have a lot of time to dedicate to this.  The PR branches should be open for new commits from you, so please see if you can progress like that.  Thanks.

Reply:
The `SMASH_SOURCE_LOCATION` fix was merged from #18 some time ago.  It seems you need to update your master, and perhaps rebase this PR on top. 

Indeed it should be `#ifdef SMASH_USE_RIVET` in `experiment.h` as you note. 

Wrt. to the problems compilng against Rivet - see the error 

```
/home/g/SMASH/smash-cholmcc/src/rivetoutput.cc: In member function ‘virtual void smash::RivetOutput::at_eventend(const smash::Particles&, int32_t, const smash::EventInfo&)’:
/home/g/SMASH/smash-cholmcc/src/rivetoutput.cc:172:19: error: cannot convert ‘HepMC3::GenEvent’ to ‘const HepMC::GenEvent&’
``` 

The compiler is complaining that it cannot convert `HepMC3::GenEvent` to `HepMC::GenEvent`.  That is, it cannot convert **HepMC3** to **HepMC**(2).  I guess you built Rivet against HepMC version 2.

Reply:
Hi! Yes, sorry, please forgive my lack of attention. Some time go I compiled Rivet both with Hepmc2 and Hepmc3, but I sourced the wrong (v2)  _rivetenv.sh_. No problems in compiling with Rivet linked to Hepmc3.